flink-staging/flink-streaming/flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/DataStream.scala

```diff
@@ -781,11 +780,24 @@ class DataStream[T](javaStream: JavaStream[T]) {
     if (fun == null) {
       throw new NullPointerException("Sink function must not be null.")
     }
+    val cleanFun = clean(fun)
     val sinkFunction = new SinkFunction[T] {
-      val cleanFun = clean(fun)
       def invoke(in: T) = cleanFun(in)
     }
     this.addSink(sinkFunction)
   }
 
+  /**
+   * Returns a "closure-cleaned" version of the given function. Cleans only if closure cleaning
+   * is not disabled in the {@link org.apache.flink.api.common.ExecutionConfig}
+   */
+  private[flink] def clean[F <: AnyRef](f: F): F = {
```

Comment:
Why do we need multiple clean functions? I think scala visibility solves that. :)